design/nfs.md

```diff
@@ -0,0 +1,157 @@
+| `serverConfig.accessMode`       | Reading and Writing permissions on the share.                             | `RO`                 |
+| `serverConfig.rootSquash`       | This prevents root users connected remotely from having root privileges. | `true`               |
+| `serverConfig.sync`             | The NFS server will not reply to requests before changes made by previous requests are written to disk. | `true` |
+| `volumes`                       | Attach a volume to NFS server Pod for sharing                            | <none>               |
+| `volumes.name`                  | Name of the volume that will be attached to NFS server pod.  | <none>               |
+| `volumes.persistentVolumeClaim` | Claim to get volume(Volume could come from hostPath, cephFS, cephRBD, googlePD, EBS etc.). | <none> |
+| `volumes.claimName`             | Name of the PVC.                                                         | <none> |
+
+### Examples:
+
+1. For sharing a volume(could be hostPath, cephFS, cephRBD, googlePD, EBS etc.) using NFS, which will be shared as /myshare by the NFS server whose Nfs-Ganesha export entry looks like:
+```
+EXPORT {
+    Export_Id = 1;
+    Path = /export;
+    Pseudo = /myshare;
+    Protocols = 4;
+    Access_Type = RW;
+    Squash = No_root_squash;
+    Sectype = sys;
+    FSAL {
+            Name = VFS;
+    }
+}
+```  
+the CRD instance will look like the following:
+```yaml
+apiVersion: rook.io/v1alpha1
+kind: NetworkFileSystem
+metadata:
+  name: mynfs
+  namespace: rook
+spec:
+  replicas: 1
+  serverConfig:
+    validHosts: all
+    accessMode: RW
+    rootSquash: false
+    sync: true
+  volumes:
+    - name: myshare
+      persistentVolumeClaim:
+        claimName: ebs-claim
+```
+
+2. For sharing multiple volumes using NFS, which will be shared as /myshare1 and /myshare2 by the NFS server whose Nfs-Ganesha export entry looks like:
+```
+EXPORT {
+    Export_Id = 1;
+    Path = /export;
+    Pseudo = /myshare1;
+    Protocols = 4;
+    Access_Type = RW;
+    Squash = No_root_squash;
+    Sectype = sys;
+    FSAL {
+            Name = VFS;
+    }
+}
+EXPORT {
+    Export_Id = 2;
+    Path = /export2;
+    Pseudo = /myshare2;
+    Protocols = 4;
+    Access_Type = RW;
+    Squash = No_root_squash;
+    Sectype = sys;
+    FSAL {
+            Name = VFS;
+    }
+}
+```  
+the CRD instance will look like the following:
+```yaml
+apiVersion: rook.io/v1alpha1
+kind: NetworkFileSystem
+metadata:
+  name: mynfs
+  namespace: rook
+spec:
+  replicas: 1
+  serverConfig:
+    validHosts: all
+    accessMode: RW
+    rootSquash: false
+    sync: true
+  volumes:
+    - name: myshare1
+      persistentVolumeClaim:
+        claimName: ebs-claim
+ volumes:
+    - name: myshare2
+      persistentVolumeClaim:
+        claimName: cephfs-claim
+```
+
+## Client Access
+Once the NFS volume is created, it can be consumed by pods using NFS PVC.  
```

Comment:
a sample yaml might be nice to show how to consume the nfs share, or else find a link that takes you straight to a sample.